src/client/ua_client.c

```diff
@@ -116,10 +116,20 @@ typedef struct {
 static UA_StatusCode
 sendSymmetricServiceRequest(UA_Client *client, const void *request,
                             const UA_DataType *requestType, UA_UInt32 *requestId) {
-    /* Make sure we have a valid session */
-    UA_StatusCode retval = UA_Client_manuallyRenewSecureChannel(client);
-    if(retval != UA_STATUSCODE_GOOD)
-        return retval;
+    UA_StatusCode retval;
+
+    /* Client in synchronous mode must call UA_Client_manuallyRenewSecureChannel
+     * in asynchronous mode UA_Client_manuallyRenewSecureChannel is called in UA_Client_runAsync */
+    if(!client->config.asyncMode) {
```

Comment:
It should be enough to test for pendingMessage.

Reply:
No really needed. Only used to remove duplicates checks. Ok I remove the code.